libsolidity/codegen/ir/Common.cpp

```diff
@@ -44,3 +44,51 @@ string IRNames::runtimeObject(ContractDefinition const& _contract)
 {
 	return _contract.name() + "_" + toString(_contract.id()) + "_deployed";
 }
+
+string IRNames::implicitConstructor(ContractDefinition const& _contract)
+{
+	return "constructor_" + _contract.name() + "_" + to_string(_contract.id());
+}
+
+string IRNames::constantValueFunction(VariableDeclaration const& _constant)
+{
+	solAssert(_constant.isConstant(), "");
+	return "constant_" + _constant.name() + "_" + to_string(_constant.id());
+}
+
+string IRNames::localVariable(VariableDeclaration const& _declaration)
+{
+	return "vloc_" + _declaration.name() + '_' + std::to_string(_declaration.id());
+}
+
+string IRNames::localVariable(Expression const& _expression)
+{
+	return "expr_" + to_string(_expression.id());
+}
+
+string IRNames::trySuccessConditionVariable(Expression const& _expression)
+{
+	auto annotation = dynamic_cast<FunctionCallAnnotation const*>(&_expression.annotation());
+	solAssert(annotation, "");
+	solAssert(annotation->tryCall, "Parameter must be a FunctionCall with tryCall-annotation set.");
+
+	return "trySuccessCondition_" + to_string(_expression.id());
+}
+
+string IRNames::tupleComponent(size_t _i)
+{
+	return "component_" + to_string(_i + 1);
+}
+
+string IRNames::zeroValue(Type const& _type, string const& _variableName)
+{
+	return "zero_value_for_type_" + _type.identifier() + _variableName;
+}
+
+string IRNames::join(string const& _baseName, string const& _suffix)
```

Comment:
This one I'm also not sure about... is this really generally useful?
It's used only once in the private function ``IRVariable::suffixedName``... I think it can either remain inlined there, or ``IRVariable::suffixedName`` could actually be removed and calls to it could be replaced directly by ``IRNames::join(m_baseName, _suffix)`` in ``IRVariable``?

I think I'd just move this one back as well for now.

Reply:
I think it might come in handy. I expected to already find a few uses for it while going through the code since concatenating two things and adding something between them only if they're both non-empty is something I often need. And usually I don't have a nice helper ready at hand and I don't feel like looking for a good global place to add one and I don't want to make the code more complex with `if`s. But looks like the code constructing names in IR generator almost always just adds the `_` unconditionally. Not sure if that's because names can't be empty in all these cases or if someone also didn't want to bother with it and decided that having a dangling `_` is not a big deal :)

Well, I'll remove it. I specifically put this and the other change in separate commits to be able to easily remove them. I even considered separate PRs but didn't want to go overboard with granularity.

Reply:
Done.